pyrevolve/storage.py

```diff
@@ -1,10 +1,95 @@
 import numpy as np
+import datetime
 from functools import reduce
 from operator import mul
 
 from .logger import logger
 from .compression import CompressedObject
 import pickle
+import os
+
+
+class DiskStorage(object):
+    """Stores all checkpoints on a .dat binary file. The
+    []-operator is overloaded to return a file object with its
+    file-pointer placed according to the checkpoint key.
+    Revolve will typically use this as LIFO, but the storage also supports
+    random access. By default, the .dat file is removed when the object
+    is destroyed.
+    """
+
+    """Creates a .dat bynary file for storing checkpoints in
+    disk. Requires number of checkpoints andsize of one checkpoint.
+    """
+    def __init__(self, size_ckp, n_ckp, dtype, profiler,
+                 filename=None, remove_file=True):
+        self.size_ckp = size_ckp
+        self.n_ckp = n_ckp
+        self.dtype = dtype
+        self.profiler = profiler
+        if filename is None:
+            self.filename = "CKP_{}.dat".format(
+                datetime.datetime.now().strftime("%Y%m%d-%H%M%S"))
+        else:
+            self.filename = filename
+        self.storage_size = 0
+        self.shapes = {}
+        self.rm_file = remove_file
+        self.storage_w = open(self.filename, 'bw+')
+        self.storage_r = open(self.filename, 'br+')
+        self.default_storage_stream = self.storage_w
+
+    def setW(self):
+        ''' Sets WRITE stream as default '''
+        self.default_storage_stream = self.storage_w
```

Comment:
What's going on here?

Reply:
"Load" and "save" methods access "read" and "write" file streams respectively, but both use the same implementation of __getitem__(). I had to switch streams explicitly to avoid messing with the file pointer (seek), because the "appy_reverse"  method in alternates between load and save calls.

Reply:
I was actually wondering about this. Why do we choose to store everything in one file? Why not one file per checkpoint, for example?